WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -102,6 +110,7 @@ final class AddEditCouponViewModel: ObservableObject {
     @Published var freeShipping: Bool
     @Published var couponRestrictionsViewModel: CouponRestrictionsViewModel
     @Published var productOrVariationIDs: [Int64]
+    @Published var categoryIDs: [Int64]
```

Comment:
What do you think about making `categoryIDs` private and introducing new read-only properties as below?

**For checking `categoryIDs` count**
```swift
var selectedProductCategories: Bool {
    categoryIDs.isNotEmpty
}
```

**For creating localized strings.** (Move relevant Localization strings to view model)
```swift
var productCategoriesButtonText: String {
    guard selectedProductCategories else {
        return Localization.selectCategoriesButton
    }

    return String.localizedStringWithFormat(Localization.editProductCategoriesButton, categoryIDs.count)
}
```

I think by doing the above we can,
1. Avoid exposing `categoryIDs`.
1. Make `AddEditCoupon` lighter.
1. Add unit tests.

Reply:
Good thought! Updated `categoryIDs` along with `productAndVariationIDs` in e9f2ef7915d3b808ac415249a44d38660a934323 - the unit tests will be updated later. A similar update was also made for `CouponRestrictionsViewModel` in e7ff24d5beb62fd2722a51bac0100b0c0e5aaf0a.